src/runtime/account_loader.zig

```diff
@@ -70,613 +77,422 @@ pub const LoadedTransactionAccounts = struct {
-            // a previous instruction deallocated this account, we will make a new one in its place.
-            var account = AccountSharedData.EMPTY;
-            account.rent_epoch = RENT_EXEMPT_RENT_EPOCH;
-
-            const account_ptr = self.account_cache.getPtr(key.*).?;
-            allocator.free(account_ptr.data);
-            account_ptr.* = account;
-
-            return LoadedTransactionAccount{
-                .account = account_ptr,
-                .loaded_size = 0,
-                .rent_collected = 0,
-            };
-        };
-
-        const rent_collected = collectRentFromAccount(
-            account.account,
-            key,
-            feature_set,
-            slot,
-            rent_collector,
-        );
-
+fn loadTransactionAccount(
+    map: *AccountMap,
+    allocator: Allocator,
+    transaction: *const RuntimeTransaction,
+    rent_collector: *const RentCollector,
+    feature_set: *const sig.core.FeatureSet,
+    slot: sig.core.Slot,
+    key: *const Pubkey,
+    is_writable: bool,
+) error{OutOfMemory}!LoadedTransactionAccount {
+    if (key.equals(&runtime.sysvar.instruction.ID)) {
+        @branchHint(.unlikely);
         return .{
-            .account = account.account,
-            .loaded_size = account.loaded_size,
-            .rent_collected = rent_collected.rent_amount,
+            .account = try constructInstructionsAccount(allocator, transaction),
+            .loaded_size = 0,
+            .rent_collected = 0,
+            .is_owned = true,
         };
     }
 
-    /// null return => account is now dead
-    pub fn loadAccount(
-        self: *BatchAccountCache,
-        allocator: Allocator,
-        transaction: *const RuntimeTransaction,
-        key: *const Pubkey,
-        is_writable: bool,
-        formalized_loaded_data_size: bool,
-    ) error{OutOfMemory}!?LoadedTransactionAccount {
-        const base_account_size: u64 = if (formalized_loaded_data_size)
-            TRANSACTION_ACCOUNT_BASE_SIZE
-        else
-            0;
-
-        const maybe_account: ?*AccountSharedData =
-            if (key.equals(&runtime.sysvar.instruction.ID)) account: {
-                @branchHint(.unlikely);
-                const account = try self.sysvar_instruction_account_datas.addOne(allocator);
-                account.* = try constructInstructionsAccount(allocator, transaction);
-                break :account account;
-            } else self.account_cache.getPtr(key.*);
-
-        const account = maybe_account orelse unreachable;
-        if (account.lamports == 0) return null;
-
-        // agave "inspects" the account here, which caches the initial state of writeable accounts
-        // TODO: we should probably do this at init time
-        _ = is_writable;
+    var account = (try loadAccount(
+        map,
+        allocator,
+        transaction,
+        key,
+        is_writable,
+        feature_set.active(.formalize_loaded_transaction_data_size, slot),
+    )) orelse {
```

Comment:
I believe you can drop the parens here?
```suggestion
    var account = try loadAccount(
        map,
        allocator,
        transaction,
        key,
        is_writable,
        feature_set.active(.formalize_loaded_transaction_data_size, slot),
    ) orelse {
```

Reply:
6a14047b106822320926032901b1d9ebbdada499